openBCIBoard.js

```diff
@@ -1,14 +1,486 @@
-var serialport = require('serialport');
-module.exports = {
-  streaming: false,
-  serial: null,
-  createBoard: function(board){
-    return new serialport.SerialPort(board.port, {
-      baudrate: board.baudrate,
-      parser: this.parseData
+'use strict';
+
+var k = require('./OpenBCIConstants');
+var OpenBCISample = require('./OpenBCISample');
+var serialPort = require('serialport');
+var EventEmitter = require('events').EventEmitter;
+var util = require('util');
+var stream = require('stream');
+
+
+
+
+function OpenBCIFactory() {
+    var factory = this;
+
+    var _options = {
+        baudrate: 115200,
+        daisy: false
+    };
+
+    /**
+     * Purpose: The initialization method to call first, before any other method.
+     * Author: AJ Keller (@pushtheworldllc)
+     */
+    function OpenBCIBoard(portName,options,connectImmediately) {
+        var self = this;
+        var args = Array.prototype.slice.call(arguments);
+
+
+
+        options = (typeof options !== 'function') && options || {};
+
+        var opts = {};
+
+        connectImmediately = (connectImmediately === undefined || connectImmediately === null) ? true : connectImmediately;
+
+        stream.Stream.call(this);
+
+        opts.baudRate = options.baudRate || options.baudrate || _options.baudrate;
+
+        opts.daisy = options.daisy || options.daisy || _options.daisy;
+
+        self.badPackets = 0;
+        self.bytesIn = 0;
+        self.options = opts;
+        self.portName = portName;
+        self.moneyBuf = new Buffer('$$$');
+        self.lookingForMoney = true;
+        self.masterBufferMaxSize = 3300;
+        // Buffer used to store bytes in and read packets from
+        self.masterBuffer = {
+            buffer: new Buffer(self.masterBufferMaxSize),
+            positionRead: 0,
+            positionWrite: 0,
+            packetsIn:0,
+            packetsRead:0,
+            looseBytes:0
+        };
+
+        //TODO: Add connect immediately functionality, suggest this to be the default...
+        if(connectImmediately) {
+            /** Step 1:  Instaniate serialport */
+
+            /** Step 2:  Wait for 'open' event on serialport */
+
+            /** Step 3:  Send a soft reset */
+
+            /** Step 4:  Wait for a '$$$' on buffer */
+
+        }
+
+    }
+
+    // This allows use to use the emitter class so freely in the index.js method
```

Comment:
Should read, "This allows us to use the emitter class freely outside of the module"